### PR TITLE
💚️ Disable CI pipeline triggers which cause unnecessary builds.

### PR DESCRIPTION
**Pull request by madchutney**
### Description

Disable CI pipeline triggers which cause unnecessary builds.

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [x]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -45,5 +45,5 @@ max-line-length = 160
 docstring-convention=google
 
 [tool:pytest]
-addopts = --cov mbed_targets --cov-report xml:coverage/coverage.xml
+addopts = --junitxml=junit/test-results.xml --cov mbed_targets --cov-report xml:coverage/coverage.xml --cov-report=html
```
Comment (muchzill4) [inline setup.cfg]: Welp, I did remove `html` generation. Guess that'll be useful for local development. 👍 

Why is `junitxml` necessary?

**Comment by muchzill4** [comment]
@madchutney Since this is OSS, I presume external contributions will come from forked repos. Is this configuration going to trigger builds on PRs that are from forks?

**Comment by madchutney** [comment]
@muchzill4 That is a good question, forks are building fine now but I will need to test once I push to master. I suspect we will have to manually trigger the build of PRs from forks. So the question will be which of the following options we want:

1. Changes to a PR always trigger **two** builds except on forks when there will be **one**.
2. Changes to a PR always trigger **one** build except on forks where it will have to be **manually** triggered.

File: setup.cfg (hunk 1 above)
Comment (madchutney) [reply setup.cfg]: It was the recommended command line for Python, thought it might help with the warnings and be generally useful.

**Pull request merged**